src/v/cluster/isolation_node_watcher.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cluster/health_monitor_frontend.h"
+#include "cluster/metadata_cache.h"
+#include "cluster/node_status_table.h"
+#include "seastarx.h"
+
+namespace cluster {
+
+class isolation_node_watcher {
```

Comment:
nit: would it make more sense to name it `node_isolation_watcher` ? @jcsp what do you think ?

Reply:
Yes, `node_isolation_watcher` would read a bit more cleanly.